relay-server/src/services/stats.rs

```diff
@@ -0,0 +1,108 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use relay_config::{Config, RelayMode};
+use relay_statsd::metric;
+use relay_system::{Addr, Service};
+use tokio::time::interval;
+
+use crate::services::upstream::{IsNetworkOutage, UpstreamRelay};
+use crate::statsd::{RelayGauges, TokioGauges};
+
+/// Relay Stats Service.
+///
+/// Service which collects stats periodically and emits them via statsd.
+pub struct RelayStats {
+    config: Arc<Config>,
+    upstream_relay: Addr<UpstreamRelay>,
+}
+
+impl RelayStats {
+    pub fn new(config: Arc<Config>, upstream_relay: Addr<UpstreamRelay>) -> Self {
+        Self {
+            config,
+            upstream_relay,
+        }
+    }
+
+    async fn tokio_metrics(&self) {
+        let m = tokio::runtime::Handle::current().metrics();
+
+        metric!(gauge(TokioGauges::ActiveTasksCount) = m.active_tasks_count() as u64);
+        metric!(gauge(TokioGauges::BlockingQueueDepth) = m.blocking_queue_depth() as u64);
+        metric!(gauge(TokioGauges::BudgetForcedYieldCount) = m.budget_forced_yield_count());
+        metric!(gauge(TokioGauges::NumBlockingThreads) = m.num_blocking_threads() as u64);
+        metric!(gauge(TokioGauges::NumIdleBlockingThreads) = m.num_idle_blocking_threads() as u64);
+        metric!(gauge(TokioGauges::NumWorkers) = m.num_workers() as u64);
+        for worker in 0..m.num_workers() {
+            let worker_name = worker.to_string();
+            metric!(
+                gauge(TokioGauges::WorkerLocalQueueDepth) =
+                    m.worker_local_queue_depth(worker) as u64,
+                worker = &worker_name,
```

Comment:
Are we OK with the cardinality this produces (`num_pods * num_workers`)?

Reply:
What's the alternative? Can we emit a custom gauge with min/max/avg?